src/store/index-level.ts

```diff
@@ -76,33 +76,36 @@ export class IndexLevel {
       throw new DwnError(DwnErrorCode.IndexMissingIndexableProperty, 'Index must include at least one valid indexable property');
     }
 
-    const indexOps: LevelWrapperBatchOperation<string>[] = [];
+    const item: IndexedItem = { messageCid, indexes };
+    const indexOpPromises: Promise<LevelWrapperBatchOperation<string>>[] = [];
 
     // create an index entry for each property index
     // these indexes are all sortable lexicographically.
+    // the key is the index property's value followed by the messageCid as a tie-breaker.
+    // for example if the property is messageTimestamp the key would look like:
+    // '"2023-05-25T18:23:29.425008Z"\u0000bafyreigs3em7lrclhntzhgvkrf75j2muk6e7ypq3lrw3ffgcpyazyw6pry'
     for (const indexName in indexes) {
       const indexValue = indexes[indexName];
-      // the key is indexValue followed by the messageCid as a tie-breaker.
-      // for example if the property is messageTimestamp the key would look like:
-      // '"2023-05-25T18:23:29.425008Z"\u0000bafyreigs3em7lrclhntzhgvkrf75j2muk6e7ypq3lrw3ffgcpyazyw6pry'
-      const key = IndexLevel.keySegmentJoin(IndexLevel.encodeValue(indexValue), messageCid);
-      const item: IndexedItem = { messageCid: messageCid, indexes };
-
-      const partitionOperation = await this.createOperationForIndexPartition(
-        tenant,
-        indexName,
-        { type: 'put', key, value: JSON.stringify(item) }
-      );
-      indexOps.push(partitionOperation);
+      if (Array.isArray(indexValue)) {
+        for (const indexValueItem of indexValue) {
+          const partitionOperationPromise = this.createPutIndexedItemOperation(tenant, item, indexName, indexValueItem);
+          indexOpPromises.push(partitionOperationPromise);
+        }
+        continue;
+      }
+
+      const partitionOperationPromise = this.createPutIndexedItemOperation(tenant, item, indexName, indexValue);
```

Comment:
Nit: This is a case where I think `else` conveys the intent (thus more readable) better than `continue`.